.github/workflows/build-check.yaml

```diff
@@ -0,0 +1,33 @@
+name: Build
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        arch: [x86_64, aarch64, loongarch64, powerpc64le, wasm]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Install Zig compiler
+        uses: mlugg/setup-zig@v1
+      - name: Install Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '~1.24.2'
+
+      - name: Install make on Windows
+        if: matrix.os == 'windows-latest'
+        run: |
+          choco install make
+        shell: powershell
+
+      - name: Run make build
+        run: make build
```

Comment:
I would also love to have some kind of test, but this is fine.

Reply:
No, that's a good idea. Can I add it in a new PR. I'd like to have a different workflow for it

Reply:
sounds good